### DISCUSSION TITLE
Foxhole World Map

### DISCUSSION THREAD

**Discussion by ghost** [mathoudebine/turing-smart-screen-python#341] [General]
A project I have been working on. Foxhole is an MMO game that does its best to simulate a 24x7 war that can last anywhere from 30-45 days. During that 45 days territory is fought over and technology is unlocked. The early days of the war are fought with bolt action rifles and grenades. By the end you're fighting with heavy machine guns, artillery and large battle tanks.

[Foxhole Trailer](https://www.youtube.com/watch?v=2EHVg4RyHvw)

The developers released an API that allows us to generate a world map using a vonoroi diagram. The world map represents the territory controlled by each faction. The API also provides some additional details like casualties, building ownership and some other goodies.

My plan is to put the project on github in the next few weeks. The developers are scheduled to showcase their next release of the game next week and were anticipating that the map will be expanded.

https://github.com/mathoudebine/turing-smart-screen-python/assets/141661840/b1c6ee1d-c99c-4b4b-8e64-b23ac7ecd305

I used the turing-smart-screen-python API to interface with the turing LCD (which was honestly the easiest part of this project) and then built a custom library to connect to the Foxhole API [FoxAPIWrapper](https://github.com/foxholenoob/FoxAPIWrapper) and a process to use the data from the API to generate the world map and some other display information. The original plan was to do this with C# but I was not impressed with the LCD libraries out there and decided to go with Python (which I had never used before).

Each hexagon of the map is generated separately and through a series of steps starting with the original game map, drawing the voronoi diagram on top and finally shading the color of the faction ownership. All 38 regions are then combined into one single map. The API is called every 30 seconds and rebuilds the map accordingly.

![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/141661840/42d467f7-0707-487a-89c0-0eff0c554451)

![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/141661840/1186e26d-481a-4aa2-b45f-bf6a1985e1c6)

![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/141661840/41b5ee32-cb7e-4f50-a28e-000925879b1a)

**Comment by mathoudebine** [comment]
Amazing! I was about to ask since I saw your last screen capture in the orientation issue.
This is a very ingenious way of using this kind of display and I'm glad you were able to integrate this Python code (especially if it was easy!) into your own code to create what you had in mind.

I have not had much feedback on people using their smart screens for other purposes than system monitoring, so thanks for taking the time to explain it here and giving some context.
I like the idea of repurposing these smart displays for other usages then what was originally planned when they were marketed - can we call it "hacking"?

I look forward to browsing your code when you upload it, especially the bitmap generation. Please keep us posted!
Can I add this post (and maybe your repository) in the Readme as example?

**Comment by ghost** [comment]
> Can I add this post (and maybe your repository) in the Readme as example?


Absolutely, the developers of Foxhole just dropped the trailer for the latest version of their game. Lots of changes to the map. If were lucky well get access to these new features this Tuesday and ill open the repository to public and ill message you.

The only thing I am struggling with is finding a retailer who offers the 5 inch model, since I wrote my code specifically around the 3.5 model.

**Comment by mathoudebine** [comment]
For the 5 inch models you can find them on Aliexpress e.g. https://www.aliexpress.us/item/3256804659136011.html or https://www.aliexpress.us/item/3256804199469843.html for the Turing 5 inch or https://www.aliexpress.us/item/3256805454873209.html for the UsbMonitor 5 inch
Both models are supported by this project.
The Turing 5 inch has additional features like gif/video background or higher refresh rate but they are not yet supported by this project.
